### DISCUSSION TITLE
Getting Started - Conditional Triggers

### DISCUSSION THREAD

**Discussion by angryjames** [theyosh/TerrariumPI#783] [Q&A] [answered]
Hey all, New to the group, hoping I can add my input down the road.

I am a bit confused about how to do a conditional-type trigger, or if it is even possible.

For Example,
During Night TIme, when the Main lights are off, I want to keep track of the temperature, and if it drops below a certain threshold, I want to kick on the CHE (Ceramic Heating Element).  then when it gets above another threshold (during the night time), I can turn it off.

I feel a Scenario is what I am looking for, but I am not 100% sure.  Any Input would be appreciated

**Comment by theyosh** [comment, marked as answer]
Hi,

Yup, the scenario is missing in the documentation. So I will try to explain it here.
You have to setup some temperature sensors for this. 1 is enough, but more is possible. When you add the sensors you specify the min and max alarm value. Those are used as triggers for relays later on. For example my temperature sensor that is controlling a heating mat.

![image](https://user-images.githubusercontent.com/8080177/219771193-1afe36db-0f93-4539-ba96-eca2800feee4.png)

Values outside the limit values are treated as broken sensor readings. So make sure you have enough room between the alarm values and the limit values. You can always adjust later on.

Next add a relay that controls your CHE. That is just plain simple adding it.

Then the fun part starts. First you have to create an enclosure. This enclosure will hold all kind of area logic in it.
In you case, as you want to heater to work only at night, you have to add a light area with the 'main lights' toggled on. This area is than used for the heating are to determine when it is night. Basically, that is when the lights are off.

Then it is time to create a heating area. In this are you select the type `heating`. Next you select the temperature sensors you want to use. If you select more than 1, the average of the sensors is used. And set the mode on `sensors`. This way the sensors will control the relay later on. I use the daylight source the lights. You could also use the weather info if you want.

![image](https://user-images.githubusercontent.com/8080177/219773526-2b2ea8a6-5043-432a-8ea3-728b9bc9fb6b.png)

Next, for you important, select the option `Off` at the light status. This means that the relay can only be on, when the main lights are off. And select the relay you want to use. This is all done in the `low alarm` tab, which is working with the low alarm value of the sensor(s). Does it get below it, it will toggle on.
The relay will toggle off when it gets **to hot**. In other words, when it reaches the high alarm value of the sensors. This will toggle off the relays until it gets below the low alarm value.
If you use a dimmer, the setup is the same, but the dimmer will work a bit different. But that does not matter for now.

I hope this will help you to get it to work you want. Also, press the small question mark button next to the title to get some help.

**Reply by angryjames** [reply]
Perfect, that is what I was going to go with.   I am still confused about one thing though.

Is the Alarm Deviation basically in lieu of the "Normal" alarm values?  Say I set the sensor up with a high alarm of 110(f). If I wanted this scenario to turn off the CHE at 75(f), would I set the high deviation to -35?
